squirtgun-platform-fabric/build.gradle.kts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2021 Lucy Poulton
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+description = "squirtgun-platform-fabric"
+
+plugins {
+    id("fabric-loom") version "0.8-SNAPSHOT"
+}
+
+val minecraftVersion = "1.17"
+val yarnBuild = 13
+val loaderVersion = "0.11.6"
+val fabricApiVersion = "0.35.2+1.17"
+val luckoFabricPermissionsApiVersion = "0.1-SNAPSHOT"
+
+repositories {
+    maven("https://oss.sonatype.org/content/repositories/snapshots/")
+}
+
+dependencies {
+    minecraft("com.mojang:minecraft:${minecraftVersion}")
+    mappings("net.fabricmc:yarn:${minecraftVersion}+build.${yarnBuild}:v2")
+    modCompileOnly("net.fabricmc:fabric-loader:${loaderVersion}")
+
+    modCompileOnly("net.fabricmc.fabric-api:fabric-api:${fabricApiVersion}")
+    modCompileOnly("me.lucko:fabric-permissions-api:${luckoFabricPermissionsApiVersion}") {
+        isTransitive = true
+    }
+
+    api(project(":squirtgun-api"))
+    api(project(":squirtgun-commands"))
+    modApi("net.kyori:adventure-platform-fabric:4.1.0-SNAPSHOT")
+}
+
+tasks {
+    javadoc {
+        val opts = options as StandardJavadocDocletOptions
+        listOf(
+            "https://maven.fabricmc.net/docs/yarn-${minecraftVersion}+build.${yarnBuild}",
+            "https://"
+        ).forEach { opts.links?.add(it) }
+    }
+
+    withType<GenerateModuleMetadata> {
+        enabled = true
+    }
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("mavenJava") {
+            artifact(tasks.remapJar) { builtBy(tasks.remapJar) }
+            artifact(tasks.sourcesJar) { builtBy(tasks.remapSourcesJar) }
+            artifact(tasks.javadocJar) { builtBy(tasks.javadocJar) }
+
+            pom {
+                artifactId = artifactId
+                description.set("A multipurpose library designed for Minecraft: Java Edition plugins.")
```

Comment:
could we maybe change this to something fabric-specific? i've been planning to do that with the other platforms too

Reply:
Sure